File: Strings/__2000__/2728/2728.cpp
#include <iostream>
#include <sstream>
#include <algorithm> // para transform()
using namespace std;

int main() {
    string line;
    while(getline(cin, line)) {
        stringstream ss(line);
        string palavra;

        string cobol = "cobol";
        bool valido = true;

        int ind = 0;
        while(getline(ss, palavra, '-')) {
            // Transforma a palavra em minúscula
            transform(palavra.begin(), palavra.end(), palavra.begin(), ::tolower);

            char letra = cobol[ind];
            if (palavra.front() != letra && palavra.back() != letra) {
                valido = false;
                break;
            }
            ind++;
        }

        cout << (valido ? "GRACE HOPPER" : "BUG") << endl;
    }

    return 0;
}

File: Strings/__2000__/2728/2728.php
<?php

while ($linha = readline()) {
    $palavras = explode("-", $linha);

    $cobol = "cobol";
    $valido = true;

    foreach ($palavras as $i => $palavra) {
        $char = $cobol[$i];
        $letraInicial = $palavra[0] ?? '';
        $letraFinal = $palavra[strlen($palavra) - 1] ?? '';

        if (strcasecmp($letraInicial, $char) !== 0 && strcasecmp($letraFinal, $char) !== 0) {
            $valido = false;
            break;
        }
    }

    echo $valido ? "GRACE HOPPER\n" : "BUG\n";
}
